core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -150,35 +82,71 @@ impl ConversionRateFetcher for NativeTokenFetcher {
                 .load(std::sync::atomic::Ordering::Relaxed),
         )
     }
+
+    async fn update(&self) -> anyhow::Result<()> {
+        match self
+            .http_client
+            .get(format!(
+                "{}/conversion_rate/0x{}",
+                &self.config.host,
+                &self.config.token_address.encode_hex::<String>()
+            ))
+            .send()
+            .await
+        {
+            Ok(response) => {
+                let conversion_rate = response.json::<u64>().await.context(
+                    "Unable to parse the response of the native token conversion rate server",
+                )?;
+                self.latest_to_eth_conversion_rate
+                    .store(conversion_rate, std::sync::atomic::Ordering::Relaxed);
+                self.error_reporter.reset();
+            }
+            Err(err) => self.error_reporter.process(anyhow::anyhow!(err)),
+        }
+
+        Ok(())
+    }
 }
 
 #[derive(Debug)]
 struct ErrorReporter {
-    current_try: u8,
-    alert_spawned: bool,
+    current_try: AtomicU8,
+    alert_spawned: AtomicBool,
```

Comment:
Nit: I think having a single atomic variable to implement interior mutability is usually fine, but two is sus (especially given that if this would ever actually be used in a shared context, it may introduce bugs). Probably the right approach here would be to put it under `Mutex` (actually, `RefCell` should be fine since we won't hold the guard across an await point, but up to you).

Reply:
After looking into it a bit, I realized the way the logic worked, the `alert_spawned` boolean could be removed, as essentially all we are doing is sending an alert when the `MAX_CONSECUTIVE_NETWORK_ERRORS` threshold is reached. 

Removed it here [6e0c812](https://github.com/matter-labs/zksync-era/pull/1360/commits/6e0c81294b7f78d85d8768356ba0cd3815c82de2) so now there's just one atomic variable.

Reply:
I'm not sure if the new code is equivalent to the old behavior. Given that the variable type is u8, with polling period of 3 seconds it'll overflow in ~13 minutes and spawn an alert again. The new logic also uses non-atomic increments through plain `u8`, which means that in debug mode it'll panic (as Rust in debug mode checks for integer overflows), which is probably not desired.

I think it's better to still keep two variables and use a lock on the structure. Given that there will be only a single consumer for this lock, it's not a subject for contention and overall it's not a performance-critical part of the code.

Reply:
Got it, I reverted the changes and went for an `Arc<Mutex>` implementation here [fbf66b1](https://github.com/matter-labs/zksync-era/pull/1360/commits/fbf66b15e11767592b05a5f4fecd82aca26edd21)